db.go

```diff
@@ -57,44 +60,153 @@ func Open(path string, mode DBMode) (*DB, error) {
 	cdb := (**C.notmuch_database_t)(unsafe.Pointer(&db.cptr))
 	cerr := C.notmuch_database_open(cpath, cmode, cdb)
 	runtime.SetFinalizer(db, func(db *DB) {
-		C.notmuch_database_destroy(db.toC())
+		C.notmuch_database_destroy(db.cptr)
 	})
 	return db, statusErr(cerr)
 }
 
+// Atomic opens an atomic transaction in the database and calls the callback.
+func (db *DB) Atomic(callback func(*DB)) error {
+	cerr := C.notmuch_database_begin_atomic(db.cptr)
+	if err := statusErr(cerr); err != nil {
+		return err
+	}
+	callback(db)
+	cerr = C.notmuch_database_end_atomic(db.cptr)
+	if err := statusErr(cerr); err != nil {
+		return err
+	}
+	return nil
```

Comment:
Everything after the callback could be collapsed to:

    return statusErr(C.notmuch_database_end_atomic(db.cptr))